sageodata-apispec.yml

```diff
@@ -2351,21 +2352,82 @@ components:
       properties:
         unitNumber:
           format: int32
+          description: A combination of the map100000Number and sequence number.
           type: integer
         nrmRegionCode:
           description: NRM region code
+          enum:
+          - Eyre
+          - Peninsula
+          - South
+          - Australian
+          - Arid
+          - Lands
+          - East
+          - Northern
+          - '&'
+          - Yorke
+          - Alinytjara
+          - Wilurara
+          - Adelaide
+          - Mt
+          - Lofty
+          - Ranges
+          - Murray-Darling
+          - Basin
+          - Kangaroo
+          - Island
           type: string
         prescribedWellArea:
+          description: Prescribed well area code.
+          enum:
+          - Musgrave
+          - Southern
+          - Basins
+          - McLaren
+          - Vale
+          - Northern
+          - Adelaide
+          - Plains
+          - Noora
+          - Padthaway
+          - Central
+          - Far
+          - North
+          - Mallee
+          - Peake
+          - Roby
+          - and
+          - Sherlock
+          - Tintinara-Coonalpyn
+          - Lower
+          - Limestone
+          - Coast
+          - Angas-Bremer
+          - Tatiara
           type: string
         prescribedWaterResourceArea:
+          description: Prescribed water resource area code.
+          enum:
+          - Eastern
+          - Mount
+          - Lofty
+          - Ranges
+          - Baroota
+          - Marne
+          - River
+          - and
+          - Saunders
+          - Creek
+          - Barossa
+          - Valley
+          - Clare
+          - Western
```

Comment:
Each word in here has now been defined as a valid enum value. For eg: 

> and, 
> Creek etc.

I would recommend using a key as the enum value and provide a description as to what it signifies.
An example would be the _testMethod_ variable in the _wcr-pumptestdata_

Reply:
Not sure what the best approach is here, these keys don't exist in SAGeoData which means they would need to be defined only in the API source code. This means we'd have to add new entries to the code when new entries are added into SAGeoData. @n8-coder thoughts on this?

Reply:
There are lookup tables in SA Geodata that have these values. for eg: _DD_NRM_REGION_, _DD_PRESC_WATER_RESOURCE_AREA_ and _DD_PRESCRIBED_WELL_AREA_.

The problem with the data in SA Geodata is that they use free text strings as their code value, which is not great.
I recommend we define a code value for each of these in the API layer.

---
Looking at the _DD_PRESC_WATER_RESOURCE_AREA_ table... these are the value defined in SA Geodata:
|Code	|	Value|
|---|---|
|Baroota	|	Baroota Prescribed Water Resources Area|
|Barossa Valley	|	Barossa Prescribed Water Resources Area|
|Clare Valley	|	Clare Valley Prescribed Water Resources Area|
|Eastern Mount Lofty Ranges	|	Eastern Mount Lofty Ranges Prescribed Water Resources Area|
|Marne River and Saunders Creek	|	Marne River and Saunders Creek Prescribed Water Resources Area|
|Western Mount Lofty Ranges	|	Western Mount Lofty Ranges Prescribed Water Resources Area|

In the API definition we could define the enums as:
`BRTA`	- Baroota
`BRSV`	- Barossa Valley
`CLRV`	- Clare Valley
`EMLR`	- Eastern Mount Lofty Ranges
`MRSC`	- Marne River and Saunders Creek
`WMLR`	- Western Mount Lofty Ranges